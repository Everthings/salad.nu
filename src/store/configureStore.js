import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import rootReducer from "./reducer";
import api from "./middleware/api";

export default function () {
  return configureStore({
    reducer: rootReducer,
    middleware: [...getDefaultMiddleware(), api],
  });
}
