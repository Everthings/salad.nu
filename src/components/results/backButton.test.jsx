import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import configureStore from "./../../store/configureStore";
import { Provider } from "react-redux";
import Theme from "./../themes/theme";
import BackButton from "./backButton";

describe("BackButton", () => {
  let store;

  beforeEach(() => {
    store = configureStore();
  });

  it("renders back button correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Theme>
            <BackButton />
          </Theme>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should handle click correctly", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Provider store={store}>
        <Theme>
          <BackButton handleClick={handleClick} />
        </Theme>
      </Provider>
    );

    const backButton = getByTestId("back-button");
    fireEvent.click(backButton);

    expect(handleClick.mock.calls.length).toEqual(1);
  });
});
