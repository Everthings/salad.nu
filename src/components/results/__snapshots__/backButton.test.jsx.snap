// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders back button correctly 1`] = `
<button
  className="sc-AxjAm cfNtnx btn btn-danger"
  data-testid="back-button"
>
  Back
</button>
`;
